services/explorer/src/pages/Holders/Holders.effects.tsx

```diff
@@ -0,0 +1,125 @@
+import { useCallback, useEffect, useState } from 'react';
+import { ColumnData } from '../../models';
+import { usePagination } from '../../hooks/usePagination';
+import { FetchWithPaginationResult, Holder, useGetTokenPrice } from '../../api';
+import { useExplorerLoader, useExplorerNetwork } from '../../hooks';
+import { toHolderTableRow } from '../../utils';
+import { useGetHolders } from '../../api/indexer/fetchHolders';
+import { useGetTotalSupply } from '../../api/indexer/fetchTotalSupply';
+import { BigNumber } from 'ethers';
+
+const cols: ColumnData[] = [
+  { path: 'rank', name: 'Rank' },
+  { path: 'address', name: 'Address' },
+  { path: 'balance', name: 'Quantity' },
+  { path: 'percentage', name: 'Percentage' },
+  { path: 'value', name: 'Value' },
+];
+
+export const useHoldersEffects = () => {
+  const { currentNetwork, indexerEndpoint } = useExplorerNetwork();
+  const { page, rowsPerPage, handleChangePage, handleChangeRowsPerPage } =
+    usePagination();
+  const { data: tokenPriceData } = useGetTokenPrice();
+  const { initLoading, finishLoading } = useExplorerLoader();
+
+  const [totalSupply, setTotalSupply] = useState<BigNumber>(BigNumber.from(0));
+  const [tableData, setTableData] = useState([]);
+  const [start, setStart] = useState<number>(0);
+  const [pagination, setPagination] = useState<FetchWithPaginationResult>({
+    start: 0,
+    limit: rowsPerPage,
+    total: 0,
+    hasNext: true,
+  });
```

Comment:
I don't think we need this. This causes a re-render

Reply:
In my understanding, we need to keep these two separate so that we fetch only the current page... We've been trying to unify it when implementing the [nodes](https://github.com/Taraxa-project/taraxa-platform/blob/b5ebb487422695e5ecf2f2c266a73a2e045ec43f/services/explorer/src/pages/Nodes/Nodes.effects.tsx#L25)  page but it runs into full-fetch. Maybe we can discuss this one over huddle?

Reply:
Sure. We can discuss it but the only thing we need here is the `start` state and the total number of rows.